utils/lit/lit/formats/dxc_hash.py

```diff
@@ -0,0 +1,142 @@
+from __future__ import absolute_import
+import os
+import subprocess
+import platform
+import filecmp
+from glob import iglob
+
+import lit.Test
+import lit.TestRunner
+import lit.util
+import lit.ShUtil as ShUtil
+from .base import TestFormat
+
+# import HashStability.py
+import importlib.util
+hs_path = os.path.join(os.path.dirname(os.path.realpath(__file__)),
+             '..', '..', '..',
+             'hct', 'HashStability.py')
+spec = importlib.util.spec_from_file_location("HashStability", hs_path)
+module = importlib.util.module_from_spec(spec)
+spec.loader.exec_module(module)
+run_hash_stablity_test = module.run_hash_stablity_test
+
+
+class DxcHashTest(TestFormat):
+    def __init__(self, test_path, dxc_path, dxa_path, working_dir):
+        self.test_path = test_path
+        self.dxc_path = dxc_path
+        self.dxa_path = dxa_path
+        self.cwd = working_dir
+
+    def getTestsInDirectory(self, testSuite, path_in_suite,
```

Comment:
This didn't work the way I thought it did, and I'm actually a bit concerned about it after our discussion in the morning meeting.

I assumed that this was traversing using the normal shell-test flow and respecting the lit local configs. It does not. I think that is wrong.

Reply:
Updated.
Caught 2 hash mismatch
    tools\clang\test\HLSLFileCheck\hlsl\control_flow\return\lifetime-markers.hlsl
    and
    tools\clang\test\HLSLFileCheck\hlsl\objects\Cbuffer\retCBV3.hlsl

Disabled these tests for now, should fix in later PR.

Reply:
The HLSLFileCheck directory has no test suffixes enabled. So those tests shouldn't be running if you're loading the local config.

Reply:
It does fail.
If cannot find local config file, it will use local config from parent directory.
The root is using local config send from getTestsInDirectory.

Reply:
HashStability has its own suffix check, it will ignore suffix in the local config.